core/trino-main/src/main/java/io/trino/operator/DirectExchangeClientStatus.java

```diff
@@ -147,7 +159,8 @@ public DirectExchangeClientStatus mergeWith(DirectExchangeClientStatus other)
                 spilledPages + other.spilledPages,
                 spilledBytes + other.spilledBytes,
                 noMoreLocations && other.noMoreLocations, // if at least one has some locations, mergee has some too
-                ImmutableList.of()); // pageBufferClientStatuses may be long, so we don't want to combine the lists
+                ImmutableList.of(), // pageBufferClientStatuses may be long, so we don't want to combine the lists
+                getRequestDuration().mergeWith(other.getRequestDuration()));
```

Comment:
add comment that this is correct as long as all clients have the same histogram (because shape of histogram doesn't change)